lib/orderbook/MatchingEngine.ts

```diff
@@ -62,8 +46,8 @@ class MatchingEngine {
     };
   }
 
-  public static getMatchingQuantity = (buyOrder: StampedOrder, sellOrder: StampedOrder): number => {
-    if (buyOrder.price >= sellOrder.price) {
+  public static getMatchingQuantity = (buyOrder?: StampedOrder, sellOrder?: StampedOrder): number => {
```

Comment:
why params are optional?

Reply:
I could maybe handle this differently by doing undefined checks before ever calling this method. But basically lines 75 and 76 define a method and line 81 calls it with `oppositeOrder` which can be `undefined` because `peek()` will return `undefined` if the queue is empty.

Reply:
I see, I thought it shouldn't be a problem because we're checking `isEmpty` before, but obviously it doesn't matter for the return type. I'd prefer that this method won't be effected by these issues, but do whatever you think is better.

Reply:
Yeah I see what you're saying, I'll refactor this a bit.